test/e2e/framework/node/resource.go

```diff
@@ -328,25 +329,48 @@ func GetPublicIps(c clientset.Interface) ([]string, error) {
 	return ips, nil
 }
 
-// GetReadySchedulableNodesOrDie addresses the common use case of getting nodes you can do work on.
+// GetReadySchedulableNodes addresses the common use case of getting nodes you can do work on.
 // 1) Needs to be schedulable.
 // 2) Needs to be ready.
 // If EITHER 1 or 2 is not true, most tests will want to ignore the node entirely.
-// TODO: remove references in framework/util.go.
-// TODO: remove "OrDie" suffix.
-func GetReadySchedulableNodesOrDie(c clientset.Interface) (nodes *v1.NodeList, err error) {
+// If there are no nodes that are both ready and schedulable, this will return an error.
+// If max is not -1, then the returned list will contain no more than that many nodes;
+// note that sometimes tests may be run against large clusters, so don't include every
+// node in your test if you don't need to.
+func GetReadySchedulableNodes(c clientset.Interface, max int) (nodes *v1.NodeList, err error) {
 	nodes, err = checkWaitListSchedulableNodes(c)
 	if err != nil {
 		return nil, fmt.Errorf("listing schedulable nodes error: %s", err)
 	}
-	// previous tests may have cause failures of some nodes. Let's skip
-	// 'Not Ready' nodes, just in case (there is no need to fail the test).
 	Filter(nodes, func(node v1.Node) bool {
 		return isNodeSchedulable(&node) && isNodeUntainted(&node)
 	})
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("there are currently no ready, schedulable nodes in the cluster")
+	}
+	if max != -1 && len(nodes.Items) > max {
```

Comment:
This is kind of a weird behavior IMO.  I would almost prefer a wrapper around this function that is used on a limited number of call sites vs. a number of calls passing -1.